packages/default-reporter/src/reporterForClient/reportLifecycleScripts.ts

```diff
@@ -264,3 +266,18 @@ function formatLine (maxWidth: number, logObj: LifecycleLog) {
 function cutLine (line: string, maxLength: number) {
   return stripAnsi(line).substr(0, maxLength)
 }
+
+function bufferByDepPath (source: Rx.Observable<LifecycleLog>) {
+  return source.pipe(
+    groupBy(data => data.depPath),
+    mergeMap(group => {
+      return group.pipe(
+        buffer(
+          group.pipe(filter(msg => 'exitCode' in msg))
```

Comment:
we group by `msg.depPath`, 

then for each group we buffer logs until in this group we get msg with `msg.exitCode` present. Then whole buffer is streamed at once.

Reply:
I think yes. If no, we should fix it.